src/mudlet.cpp

```diff
@@ -4442,62 +4442,36 @@ bool mudlet::saveDictionary(const QString& pathFileBaseName, QSet<QString>& word
 
 QPair<bool, bool> mudlet::addWordToSet(const QString& word)
 {
-    if (mDictionaryReadWriteLock.tryLockForWrite(100)) {
```

Comment:
The user dictionary that is shared between profiles can be accessed at any time I believe - in response to, say, the user right-clicking in a profile's command line which brings up a list of words close to what is typed if that is not matched in the hunspell dictionaries that are active for the profile. As I understand that it may not be done from the same thread as the main Mudlet one - for safety's sake I included this read/write lock to protect from one or more parts of the code from trying to read the `mWordSet_shared` whilst another part is changing that set. Are you very certain that this cannot happen and thus the lock is redundant?

Reply:
Short answer: yes. Long answer, see previous replies.